wicketd/src/artifacts/update_plan.rs

```diff
@@ -529,11 +560,80 @@ impl<'a> UpdatePlanBuilder<'a> {
         Ok(())
     }
 
-    // RoT, host OS, and trampoline OS artifacts all contain a pair of artifacts
-    // we actually care about (RoT: A/B images; host/trampoline: phase1/phase2
-    // images). This method is a helper that converts a single artifact `reader`
-    // into a pair of extracted artifacts.
+    /// A helper that converts a single artifact `stream` into a pair of
+    /// extracted artifacts.
+    ///
+    /// RoT, host OS, and trampoline OS artifacts all contain a pair of
+    /// artifacts we actually care about (RoT: A/B images; host/trampoline:
+    /// phase1/phase2 images). This method is a helper to extract that.
+    ///
+    /// This method uses a `block_in_place` into synchronous code, because the
+    /// value of changing tufaceous to do async tarball extraction is honestly
+    /// pretty dubious.
+    ///
+    /// The main costs of this are that:
+    /// 1. This code can only be used with multithreaded Tokio executors. (This
+    ///    is OK for production, but does require that our tests use `flavor =
+    ///    "multi_thread`.)
```

Comment:
Ah, this answers my question above about the test flavor. I'm tempted to suggest adding a comment on the tests that use an explicit flavor pointing them here, but that seems kind of annoying since it's any test that ends up using this function. Pretty unfortunate leaky abstraction here, but I have no suggestions really, unless you think it's worth small notes on all the tests redirecting to this (very good) explanation.

Reply:
Yeah I've added small notes pointing to this documentation.